contracts/base/Multicall.sol

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.19;
+
+import {IMulticall} from "../interfaces/IMulticall.sol";
+
+/// @title Multicall
+/// @notice Enables calling multiple methods in a single call to the contract
+abstract contract Multicall is IMulticall {
```

Comment:
Ooo I've actually implemented another version of this multicall in this PR https://github.com/Uniswap/UniswapX/pull/202/files#diff-f83ec168220cc728379adbb931f4151db29f407d9cfa1130403782bcc251a6bf but added support for bubbling up custom errors

Reply:
ah ty is it just adding these few lines?:

```
if (!success) {
 // handle custom errors
  if (result.length == 4) {
      assembly {
          revert(add(result, 0x20), mload(result))
      }
  }

...
}
```

Reply:
ya should be